src/hooks/useApproveNftTransaction.ts

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+
+import { useAppSelector } from '../redux/hooks';
+import { selectApprovals, SubmittedApproval } from '../redux/stores/transactions/transactionsSlice';
+
+const useApproveNftTransaction = (): SubmittedApproval | undefined => {
+  const { collectionToken } = useAppSelector(state => state.config);
+  const pendingTransactions = useAppSelector(selectApprovals);
+
+  return useMemo(() => {
+    if (!pendingTransactions.length) {
```

Comment:
As above

```suggestion
    if (!pendingTransactions.length) return;
```

Reply:
Same, this hook returns `SubmittedApproval` or `undefined`.